setup-commit-signing/main.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+set -euo pipefail
+
+SIGNING_KEY="${1}"
+PASSPHRASE="${2}"
+
+export GNUPGHOME="$HOME/.gnupg/"
+mkdir -p "$GNUPGHOME"
+chmod 0700 "$GNUPGHOME"
+
+git config --global gpg.program /usr/bin/gpg
```

Comment:
Might want to check this exists and error otherwise. Also might want to use one from the `PATH` for macOS.